### PR TITLE
[MU4] fix #8507 Fix crashes when searching palettes/closing score

### LINKED ISSUE #8507: [MU4 Issue] Crashes searching for palettes and closing scores
**Describe the bug**
Current MU4 build crashes when searching for palettes and when closing a score

**To Reproduce**
1. Create a new score
2. Search for a palette, or close the score (using X button)
3. Crash

**Expected behavior**
Preferably no crash

**Screenshots**

**Desktop (please complete the following information):**
Windows 10

**Additional context**
Stack trace for first crash was in 
BracketItem::setProperty/Score::setLayoutAll - BracketItem was invalid
For second was 
PlaybackController::setCurrentSequence() - notationPlayback() was empty (null).

### PR DESCRIPTION
**Pull request by wizofaus**
…and ensure Ctrl+X shortcut works for slur.

Resolves: https://github.com/musescore/MuseScore/issues/8507

BracketItem lifetime management was messed up
PlaybackController wasn't handling case when NotationPlayback was null/empty
Slur was swallowing all shortcut combinations using X (or Home), so couldn't cut etc.

- [ x ] I signed [CLA](https://musescore.org/en/cla)
- [ x ] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [ x ] I made sure the code compiles on my machine
- [ x ] I made sure there are no unnecessary changes in the code
- [ x ] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [ x ] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [ x ] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/palette/internal/palette/palettecreator.cpp [hunk 1]
```diff
@@ -1099,11 +1099,15 @@ PalettePanel* PaletteCreator::newBracketsPalettePanel()
             { BracketType::LINE,   QT_TRANSLATE_NOOP("palette", "Line") } }
     };
 
-    for (auto type : types) {
+    static Staff bracketItemOwner(nullptr);
```
Comment (wizofaus) [inline src/palette/internal/palette/palettecreator.cpp:1102]: I would have preferred to use a Staff belonging to gscore - but gscore doesn't have any staves and I don't know what the consequences of adding one might be.

File: src/palette/internal/palette/palettecreator.cpp [hunk 2]
```diff
@@ -1099,11 +1099,15 @@ PalettePanel* PaletteCreator::newBracketsPalettePanel()
...
+    bracketItemOwner.setBracketType(types.size() - 1, BracketType::NORMAL);
```
Comment (wizofaus) [inline src/palette/internal/palette/palettecreator.cpp:1103]: this seems to be the correct way to ensure creation/deletion of BracketItem's is handled entirely by their parent (Staff).

File: src/engraving/libmscore/staff.cpp [hunk 3]
```diff
@@ -65,6 +65,15 @@ Staff::Staff(Score* score)
     initFromStaffType(0);
 }
 
+//---------------------------------------------------------
+//   ~Staff
+//---------------------------------------------------------
+
+Staff::~Staff()
+{
+    qDeleteAll(brackets());
```
Comment (wizofaus) [inline src/engraving/libmscore/staff.cpp:74]: memory leak

**Review by igorkorsukov — APPROVED** [review]

**Comment by Eism** [comment]
@wizofaus I still get a crash when searching pallets. Just open the score and press the search button on palette panel
[River_Flows_In_You.zip](https://github.com/musescore/MuseScore/files/6764873/River_Flows_In_You.zip)

**Comment by wizofaus** [comment]
Hmm, yep, still crashing for me too, investigating!

**Pull request merged**